osf/models/provider.py

```diff
@@ -151,6 +163,11 @@ def absolute_api_v2_url(self):
         path = '/providers/registrations/{}/'.format(self._id)
         return api_v2_url(path)
 
+    def validate_schema(self, schema):
+        if not self.schemas.filter(id=schema.id).exists():
```

Comment:
nit: I know this actually criticizing myself, but really this function would be easier to understand if it was `if not self.allowed_schemas.filter(id=schema.id).exists():` I'd consider refactoring `schemas` to `allowed_schemas` everywhere.

Reply:
I believe this was discussed in a related branded registries PR and decided against because of what front-end is expecting.